tsl/src/compression/compression_dml.c

```diff
@@ -792,8 +855,24 @@ compressed_insert_key_columns(Relation relation)
 		}
 		index_close(indexDesc, AccessShareLock);
 
-		shared_attrs = shared_attrs ? bms_intersect(idx_attrs, shared_attrs) : idx_attrs;
+		if (!shared_attrs)
+		{
+			/* First iteration */
+			shared_attrs = idx_attrs;
+			/* We only optimize unique constraint checks for non-partial indexes. */
+			*covering = indexDesc->rd_indpred == NIL;
```

Comment:
Can we use partial indexes at all for dml decompression? `build_index_scankeys_using_slot` thinks no, so maybe we shouldn't consider them here as well?

Reply:
we could at some point by testing whether the predicate applies but that is not part of this pr, but we still need to detect whether any are present